include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -1041,15 +1416,73 @@ class __brick_set_op
     }
 };
 
-template <typename _ExecutionPolicy, typename _DiffType>
+template <typename _ExecutionPolicy, typename _DiffType, typename _Range>
 struct __brick_shift_left
 {
+  private:
+    using _ValueType = oneapi::dpl::__internal::__value_t<_Range>;
+    // Maximum size supported by compilers to generate vector instructions
+    constexpr static std::uint8_t __max_vector_size = 4;
+
+  public:
+    // Multiple iterations per item are manually processed in the brick with a nd-range strided approach.
+    constexpr static std::uint8_t __preferred_iters_per_item = 1;
+    constexpr static bool __can_vectorize =
+        oneapi::dpl::__ranges::__is_vectorizable_range<std::decay_t<_Range>>::value &&
+        std::is_fundamental_v<_ValueType> && sizeof(_ValueType) < 4;
+    constexpr static std::uint8_t __preferred_vector_size =
+        __can_vectorize ? oneapi::dpl::__internal::__dpl_ceiling_div(__max_vector_size, sizeof(_ValueType)) : 1;
+
     _DiffType __size;
     _DiffType __n;
 
-    template <typename _ItemId, typename _Range>
+    template <typename _IsFull, typename _ItemId>
+    void
+    __vector_path_impl(_IsFull __is_full, const _ItemId __idx, _Range __rng) const
+    {
+        const std::size_t __unsigned_size = __size;
+        const _DiffType __i = __idx - __n;
+        oneapi::dpl::__par_backend_hetero::__vector_load<__preferred_vector_size> __vec_load{__unsigned_size};
+        oneapi::dpl::__par_backend_hetero::__vector_store<__preferred_vector_size> __vec_store{__unsigned_size};
+        oneapi::dpl::__par_backend_hetero::__scalar_load_op __load_op;
+        oneapi::dpl::__par_backend_hetero::__scalar_store_transform_op<oneapi::dpl::__internal::__pstl_assign>
+            __store_op;
+        for (_DiffType __k = __n; __k < __size; __k += __n)
+        {
+            const _DiffType __read_offset = __k + __idx;
+            const _DiffType __write_offset = __k + __i;
+            if constexpr (_IsFull::value)
+            {
+                if (__k + __idx + __preferred_vector_size <= __size)
+                {
+                    _ValueType __rng_vector[__preferred_vector_size];
+                    __vec_load(std::true_type{}, __read_offset, __load_op, __rng, __rng_vector);
+                    __vec_store(std::true_type{}, __write_offset, __store_op, __rng_vector, __rng);
+                }
+                else if (__k + __idx < __size)
```

Comment:
`__k + __idx` is already calculated above as `__read_offset`

Reply:
Addressed